lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -39,6 +39,16 @@ def default_priority
     self.priority = MiqAeDomain.highest_priority(tenant) + 1 unless priority
   end
 
+  def lock
+    self.system = true
+    save
```

Comment:
Perhaps `locked=` and set true or false?  Then you don't need the lock/unlock pair